enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -2650,9 +2654,22 @@ func (e *partitionEvictor) generateSamplesForEviction(quitChan chan struct{}) er
 			case e.samples <- sample:
 			case <-quitChan:
 				return nil
+			default:
+				// e.samples is full. Let's close the iter first, so that zombie
+				// tables can be cleaned up by pebble.
+				iter.Close()
+				shouldCreateNewIter = true
+				iter = nil
```

Comment:
could `shouldCreateNewIter` be replaced by an `iter == nil` check?

Reply:
There are other conditions where we need to refresh the iter in the for loop.